src/index.tsx

```diff
@@ -113,60 +114,60 @@ const apolloClient = new ApolloClient({
       return defaultDataIdFromObject(obj);
     }
   }),
-  link: authLink.concat(link)
+  link
+});
+
+const saleorClient = createSaleorClient({
+  apiUrl: API_URI,
+  channel: "",
+  autologin: true
 });
 
 const App: React.FC = () => (
-  <ApolloProvider client={apolloClient}>
-    <BrowserRouter basename={APP_MOUNT_URI}>
-      <ThemeProvider overrides={themeOverrides}>
-        <DateProvider>
-          <LocaleProvider>
-            <MessageManagerProvider>
-              <ServiceWorker />
-              <BackgroundTasksProvider>
-                <AppStateProvider>
-                  <AuthProvider>
-                    <ShopProvider>
-                      <AppChannelProvider>
-                        <ExternalAppProvider>
-                          <Routes />
-                        </ExternalAppProvider>
-                      </AppChannelProvider>
-                    </ShopProvider>
-                  </AuthProvider>
-                </AppStateProvider>
-              </BackgroundTasksProvider>
-            </MessageManagerProvider>
-          </LocaleProvider>
-        </DateProvider>
-      </ThemeProvider>
-    </BrowserRouter>
-  </ApolloProvider>
+  <SaleorProvider client={saleorClient}>
+    <ApolloProvider client={apolloClient}>
+      <BrowserRouter basename={APP_MOUNT_URI}>
+        <ThemeProvider overrides={themeOverrides}>
+          <DateProvider>
+            <LocaleProvider>
+              <MessageManagerProvider>
+                <ServiceWorker />
+                <BackgroundTasksProvider>
+                  <AppStateProvider>
+                    <AuthProvider>
+                      <ShopProvider>
+                        <AppChannelProvider
+                          onChannelChange={saleorClient.config.setChannel}
+                        >
+                          <ExternalAppProvider>
+                            <Routes />
+                          </ExternalAppProvider>
+                        </AppChannelProvider>
+                      </ShopProvider>
+                    </AuthProvider>
+                  </AppStateProvider>
+                </BackgroundTasksProvider>
+              </MessageManagerProvider>
+            </LocaleProvider>
+          </DateProvider>
+        </ThemeProvider>
+      </BrowserRouter>
+    </ApolloProvider>
+  </SaleorProvider>
 );
 
 const Routes: React.FC = () => {
   const intl = useIntl();
   const [, dispatchAppState] = useAppState();
-  const {
-    hasToken,
-    isAuthenticated,
-    tokenAuthLoading,
-    tokenVerifyLoading
-  } = useAuth();
+  const { authenticated, authenticating } = useAuth();
 
   const { channel } = useAppChannel(false);
 
   const channelLoaded = typeof channel !== "undefined";
 
-  const homePageLoaded =
-    channelLoaded &&
-    isAuthenticated &&
-    !tokenAuthLoading &&
-    !tokenVerifyLoading;
+  const homePageLoaded = channelLoaded && authenticated && !authenticating;
```

Comment:
`authenticated` and `authenticating` are mutually exclusive. They can never both be `true`, so you might just want to skip the last check.